expan/core/experiment.py

```diff
@@ -274,8 +279,8 @@ def _quantile_filtering(self, kpis, percentile, threshold_type):
         :rtype: pd.Series
         """
         method_table = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
-        flags = pd.Series(data=[False]*len(self.data))
-        for column in self.data[kpis].columns:
-            threshold = np.percentile(self.data[column], percentile)
-            flags = flags | self.data[column].apply(method_table[threshold_type])
+        flags = pd.Series(data=[False]*len(data))
+        for column in data[kpis].columns:
+            threshold = np.percentile(data[column], percentile)
```

Comment:
Are you sure that `threshold` in the lambda is properly updated here? Maybe pass it along with `x` to be sure?